### ISSUE TITLE
FutureWarning in Pandas 0.24.0

### ISSUE THREAD

**Issue by iaindillingham** [altair-viz/altair#1314]
Pandas 0.24.0 throws a `FutureWarning` when used with Altair 2.3.0 [here][1], which prevents the chart being rendered correctly. The message:

```
[my_virtualenv]/lib/python3.6/site-packages/altair/utils/core.py:294: FutureWarning: A future version of pandas will default to `skipna=True`. To silence this warning, pass `skipna=True|False` explicitly.
```

I notice that Altair's `requirements.txt` doesn't specify version. Would it be worth doing so?

[1]: https://github.com/altair-viz/altair/blob/v2.3.0/altair/utils/core.py#L294

**Comment by afonit** [comment]
@iaindillingham I just upgraded to pandas 0.24.0 and am using altair 2.3.0.
I am not getting the error you are seeing, my plots are showing up fine.

What platform are you on?
Did you get the error upon import of altair or alter when calling a plot?

**Comment by iaindillingham** [comment]
Python 3.6 on Ubuntu 18.04. The error was raised when plotting data, not upon import. I'll try and reproduce.

**Comment by afonit** [comment]
I have now gotten the error.  
Interesting - it is happening on some of my plots and not on others.  I am trying to narrow it down.

**Comment by afonit** [comment]
@iaindillingham 
So far I can get it to come up if I am not declaring types for the fields.

For example:

x='actual'
vs
x='actual:Q'

Is that the case for you - that if you declare the type you don't get the error?

**Comment by jakevdp** [comment]
It looks like the warning is in pandas [infer_dtype](https://github.com/pandas-dev/pandas/blob/95f8dca47723192ddeed5a2f198d0521c687c9aa/pandas/_libs/lib.pyx#L1101) function, which we use in automatically determining the data type when not specified by the user [here](https://github.com/altair-viz/altair/blob/c4495f965db621e8bd0c2b3dd0390ef30ad33456/altair/utils/core.py#L72).

**Comment by iaindillingham** [comment]
Yes, @afonit that's correct. Although only when I use the shorthand syntax, not when I supply a `type` kwarg.

**Comment by jakevdp** [comment]
Fix is to explicitly pass ``skipna=False`` in this line: https://github.com/altair-viz/altair/blob/c4495f965db621e8bd0c2b3dd0390ef30ad33456/altair/utils/core.py#L72

We'll have to do it carefully, though, because prior to pandas 0.20.0, ``infer_dtype`` takes no keyword arguments.

**Comment by jakevdp** [comment]
I pushed a fix in #1315
